data/transactions/logic/assembler.go

```diff
@@ -1516,6 +1521,15 @@ func typePushInts(pgm *ProgramKnowledge, args []string) (StackTypes, StackTypes,
 	return nil, types, nil
 }
 
+func typePushInt(pgm *ProgramKnowledge, args []string) (StackTypes, StackTypes, error) {
+	types := make(StackTypes, len(args))
+	for i := range types {
+		val, _ := strconv.ParseUint(args[i], 10, 64)
+		types[i] = NewStackType(avmUint64, bound(val, val))
```

Comment:
I was going to write this:
```
This is really good example of why `type*` functions are hard to write. `args` may contain things like `pay` or `NoOp`, which won't be parsed correctly here. Very tedious to replicate the...
```
but looking at `asmPushInt`, it seems like we _don't_ handle constants like we do for `int`? Yuck.

Reply:
hm, so does this mean i _shouldnt_ try to parse out an integer constant from the value?

Reply:
I think what you're doing is mostly fine, but it sure seems like we _ought_ to support constants in the asmPushint function. And if we do that, then this _won't_ be fine. We just want to make sure we revisit this if we do that. So I think you _should_ react to the error from parsing here, so that if/when that happens, you'll fail rather than return `bound(0,0)` silently.